Studies/source/MvaTraining.cxx

```diff
@@ -254,41 +242,50 @@ class MVATraining {
         for(const auto& method : evaluation){
             for (const auto& sample : method.second){
-                std::map<int, double*> ks_vector;
-                std::map<int, size_t> dim;
+                std::map<size_t, std::vector<double>> ks_vector;
                 for (auto tvt : sample.second){
-                    auto type = static_cast<int>(tvt.first);
+                    auto type = tvt.first;
                     std::sort(tvt.second.begin(), tvt.second.end());
-                    std::cout<<tvt.second.front()<<"    "<<sample.first.mass<<"    "<<tvt.second.size()<<" "<<tvt.second.back()<<std::endl;
-                    dim[type] = tvt.second.size();
-                    ks_vector[type] = tvt.second.data();
+                    ks_vector[type] = std::move(tvt.second);
                 }
-                double ks = TMath::KolmogorovTest(static_cast<int>(dim.at(0)), ks_vector.at(0), static_cast<int>(dim.at(1)), ks_vector.at(1), "M");
+                double ks = TMath::KolmogorovTest(static_cast<int>(ks_vector.at(0).size()), ks_vector.at(0).data(),
+                                                  static_cast<int>(ks_vector.at(1).size()), ks_vector.at(1).data(), "");
                 kolmogorov[method.first][sample.first] = ks;
+                histo_kolmogorov->Fill(ks);
             }
         }
+
+        root_ext::WriteObject(*histo_kolmogorov, directory);
         return kolmogorov;
     }
 
-    std::map<std::string, std::pair<double, double>> Significativity(const std::map<std::string, std::map<SampleId, std::shared_ptr<TH1D>>>& outputBDT)
+    std::map<std::string, std::pair<double, double>> EstimateSignificativity(const std::map<std::string, std::map<SampleId, std::shared_ptr<TH1D>>>& outputBDT)
     {
         std::map<std::string, std::pair<double, double>> sign;
-        std::map<std::string, std::shared_ptr<TH1D>> histo_sign;
+        std::map<std::string, std::shared_ptr<TGraph>> histo_sign;
         auto directory = root_ext::GetDirectory(*outfile.get(), "Significance");
+        int nbin = 200;
         for(const auto& method : outputBDT){
             std::vector<std::pair<double,double>> cuts;
-            histo_sign[method.first] = std::make_shared<TH1D>((method.first+"_significance").c_str(), (method.first+"_significance").c_str(), 200, -1,1);
-            histo_sign[method.first]->SetXTitle("output BDT");
-            histo_sign[method.first]->SetXTitle("S/(sqrt(B))");
-            for(int i = 0; i<=200; i++){
+            histo_sign[method.first] = std::make_shared<TGraph>();
+            histo_sign[method.first]->SetTitle((method.first+"_significance").c_str());
+            histo_sign[method.first]->SetName((method.first+"_significance").c_str());
+            //            histo_sign[method.first]->SetXTitle("output BDT");
+//            histo_sign[method.first]->SetYTitle("S/(sqrt(B))");
+            std::cout<<method.first<<std::endl;
+            for(int i = 0; i<=nbin; i++){
```

Comment:
Check that number of bins in histograms = nbins. And, for safety, include overflow bin in the integral.